class.jetpack-cli.php

```diff
@@ -757,6 +757,55 @@ public function sync_queue( $args, $assoc_args ) {
 	}
 
 	/**
+	 * Cancel's the current Jetpack plan granted by this partner, if applicable
+	 *
+	 * Returns success or error JSON
+	 *
+	 * <token_json>
+	 * : JSON blob of WPCOM API token
+	 */
+	public function partner_cancel( $args, $named_args ) {
+		list( $token_json ) = $args;
+
+		if ( ! $token_json || ! ( $token = json_decode( $token_json ) ) ) {
+			$this->partner_provision_error( new WP_Error( 'missing_access_token',  sprintf( __( 'Invalid token JSON: %s', 'jetpack' ), $token_json ) ) );
+		}
+
+		if ( isset( $token->error ) ) {
+			$this->partner_provision_error( new WP_Error( $token->error, $token->message ) );
+		}
+
+		if ( ! isset( $token->access_token ) ) {
+			$this->partner_provision_error( new WP_Error( 'missing_access_token', __( 'Missing or invalid access token', 'jetpack' ) ) );
+		}
+
+		$blog_id    = Jetpack_Options::get_option( 'id' );
+
+		if ( ! $blog_id ) {
+			$this->partner_provision_error( new WP_Error( 'site_not_registered',  __( 'This site is not connected to Jetpack', 'jetpack' ) ) );
+		}
+
+		$request = array(
+			'headers' => array(
+				'Authorization' => "Bearer " . $token->access_token,
+				'Host'          => defined( 'JETPACK__WPCOM_JSON_API_HOST_HEADER' ) ? JETPACK__WPCOM_JSON_API_HOST_HEADER : 'public-api.wordpress.com',
+			),
+			'method'  => 'POST',
+			'body'    => json_encode( array( 'site_id' => $blog_id ) )
+		);
+
+		$url = sprintf( 'https://%s/rest/v1.3/jpphp/%d/partner-cancel', $this->get_api_host(), $blog_id );
+
+		$result = Jetpack_Client::_wp_remote_request( $url, $request );
+
+		if ( is_wp_error( $result ) ) {
+			$this->partner_provision_error( $result );
+		}
+
+		WP_CLI::log( print_r( $result, 1 ) );
```

Comment:
This caused a lot of extra lines when I cancelled a plan, which I didn't notice in provisioning. We may want to consider JSON stringifying the output.